Made changes to roadmap (added graphics, new info, and changed wording) (#994)
CHTC/chtc-website-source: _uw-research-computing/htc-roadmap.md

Reviewer: ```suggestion
High Throughput Computing systems specialize in running tens to millions of small, independent jobs. On the other hand, High Performance Computing systems specialize in running a few, very large jobs that use multiple computers working together on the same problem.
```

Reviewer: ```suggestion
      <td>Example tasks</td>
```

{:.tip-header}

> Rather than submitting **one large job**, consider splitting your workflow into **multiple smaller, independent jobs whenever possible**. This often leads to faster scheduling and more efficient use of HTC resources.
> <p style="text-align:center"><img src="/images/roadmap-split-workflows.png" alt="On the left, there is a complete puzzle representing one large job. On the right, those same puzzle pieces are seperated representing multiple small jobs." width=500px></p>
{:.tip}

Unlike the High Performance System, CHTC staff do not limit the number of jobs a user can have running in parallel, thus it is to your advantage to strategize your workflow to take advantage of as many compute resources as possible.

More detailed information regarding CHTC's HTC system can be found in the <a href="https://chtc.cs.wisc.edu/uw-research-computing/htc-overview">⚙️ HTC Overview Guide</a>.

> ### Step Two: Set Up Your Account and Submit Your First Job

Reviewer: I think we should keep the logging in and submitting jobs steps separate. Originally, we wanted to merge them because the logging in section was quite short initially, but with your changes and the cool new graphic, I think it can stand in its own section.

<p style="text-align:center"><img src="/images/roadmap-job-components.png" alt="Components of a job including software environment, command or script, input/output files, standard output/error, resource requirement" width=900px></p>

Jobs are submitted to **HTCondor**, the system that manages where and when jobs run. **HTCondor is a job scheduler**.

You will use **HTCondor** commands to submit jobs, check their status, review job information, and remove jobs if needed. These commands help you manage your work after it has been sent to the system.

Practicing with a simple HTCondor job first can help you feel more prepared before starting your own project.

**It is highly recommended that every user follow this short tutorial.**

<div class="d-flex mb-3">
	<div class="p-3 m-auto">
		<a class="btn btn-primary" href="https://chtc.cs.wisc.edu/uw-research-computing/htcondor-job-submission">Practice: Submit Jobs Using HTCondor</a>
	</div>
</div>

Once you are comfortable with these basic steps, you can learn how to use HTCondor to run many jobs at once.

> The tutorial below shows how to run many jobs from one submit file, use different input files or settings for each job, and save output files in specific directories.

Reviewer: ```suggestion
The following guide demonstrates different examples on how to run multiple jobs from one submit file, use different input files or settings for each job, and save output files in specific directories.
```

<div class="d-flex mb-3">
	<div class="p-3 m-auto">
		<a class="btn btn-primary" href="https://chtc.cs.wisc.edu/uw-research-computing/multiple-jobs">Practice: Submit Multiple Jobs Using HTCondor</a>
	</div>
</div>

### Step Three: Install Software

> Our <a href="https://chtc.cs.wisc.edu/uw-research-computing/htc/guides.html">Software Solutions</a> guides contain information about how to install and use software on the HTC system.

Reviewer: ```suggestion
Our <a href="htc/guides#software">software</a> guides contain information about how to install and use software on the HTC system.
```

> Before choosing how to install your software, it helps to understand the main options available on CHTC. One common option is to use a **container.**

Reviewer: ```suggestion
For most users, we recommend using a **container.**
```

**Software Containers**

> A **software container** packages your software with the tools and settings it needs to run. Some containers even include their own operating system. This helps your job run in more places on CHTC or the OSPool.

Reviewer: ```suggestion
A **software container** packages your software with the operating system, tools, and settings it needs to run into a portable object. When your job runs, it launches inside of the container with the software environment that you've set up. This ensures that your jobs can run on almost any machine, no matter which operating system or libraries the machines have installed, because your container has the tools your jobs need.
```

<p style="text-align:center"><img src="/images/roadmap-containers.png" alt="Icons for $PATH, libraries, software, code are packaged into a box representing a container" width=800px></p>

<p style="text-align:center"><em>Containers keep everything together so it is easier to move and run!</em></p>

In general, we recommend using a **container** if your software:

- Needs a specific version of R or Python
- Can be installed with <code>conda</code>
- Relies on several other tools or packages (dependencies)
- Already has a pre-existing container

> 

Reviewer: ```suggestion
Containers are great for researchers whose software stack:

- Uses Python, R, Julia, and MATLAB
- Uses conda
- Rely on several other tools or packages (dependencies)
- Have pre-existing containers (such as on [Docker Hub](hub.docker.com/)

```

> Using a container can make your work more portable and easier to run consistently. Learn more through the guide below that explains how to build, test, and use software containers.

Reviewer: ```suggestion
Using a container makes your work more portable and easier to run consistently. Learn more through the guide below that explains how to build, test, and use software containers.
```

<div class="d-flex mb-3">
	<div class="p-3 m-auto">
		<a class="btn btn-primary" href="https://chtc.cs.wisc.edu/uw-research-computing/software-overview-htc#using-containers-on-chtc">Using containers on CHTC</a>
	</div>
</div>

> Containers are a good option for many workflows, but they are not the only way to use software on CHTC. Some software is already installed on CHTC and can be used through **modules**.

Reviewer: ```suggestion
```

This sentence sounds a bit like we'll install software in modules for researchers. This is not usually the case, so I'll remove this entirely.

**Pre-installed Software in Modules**

> CHTC provides a limited set of pre-installed software called modules. You can load these modules and use them in your jobs.

Reviewer: ```suggestion
CHTC provides a limited set of pre-installed software in modules. You can load these modules and use them in your jobs.
```

Available modules include software used in several research areas, such as COMSOL, ANSYS, ABAQUS, GUROBI, and others.

> To learn how to use this software in your jobs, see the <a href="htc-modules">💡 Use Software Available in Modules</a> guide and the <a href="licensed-software">👾 Use Licensed Software</a> guide.

Reviewer: ```suggestion
To learn how to use these software in your jobs, see the <a href="htc-modules">💡 Use Software Available in Modules</a> guide and the <a href="licensed-software">👾 Use Licensed Software</a> guide.
```

### Step Four: Access your Data

When you start using the HTC system, you will usually need to upload your data files to CHTC. This allows your jobs to use the files while they run.

If you do not want to upload your data to CHTC, you may be able to set up your HTCondor jobs to transfer files in other ways. For example, your jobs can:

- Pull or push files using <code>s3</code> file transfer
- Download files using standard Unix commands, such as <code>wget</code>
- Use other file transfer methods

> 

Reviewer: ```suggestion
```

I think this may be a bit too much information for now, so let's delete.

The best way to move your data depends on where your files are stored and how your workflow is set up. The guides below explain how to transfer files from several common locations:

- <a href="transfer-files-computer">Transfer Files between CHTC and your Computer</a>
- <a href="transfer-data-researchdrive">Transferring Files Between CHTC and ResearchDrive</a>
- <a href="globus">Using Globus to Transfer Files to and from CHTC</a>
- <a href="github-remote-access">Remotely Access a Private GitHub Repository</a>

> <p style="text-align:center"><img src="/images/roadmap-transfer-data.png" alt="Data can move to CHTC from your computer, ResearchDrive, GitHub, or Globus using different transfer tools, then be saved in /home or /staging" width=800px></p>

Reviewer: I just remembered we forgot to add the "Internet" and "--> wget, curl" to the diagram! We don't specifically have a guide for that yet, but can you add this to the illustration? This can be `/home` or `/staging`.

**Choose a Location to Stage your Data**

When uploading data to the HTC system, users need to choose a location to store that data on our system. There are two primary locations: <code>/home</code> and <code>/staging</code>. 

> <code>/home</code> is more efficient at handling "small" files, while <code>/staging</code> is more efficient at handling "large" files. For more information on what is considered "small" and "large" data files and to learn how to use files stored in these locations for jobs, visit our <a href="guides">📁 HTC Data guides</a> . 

Reviewer: ```suggestion
<code>/home</code> is more efficient at handling "small" files, while <code>/staging</code> is more efficient at handling "large" files. For more information on what is considered "small" and "large" data files and to learn how to use files stored in these locations for jobs, visit our <a href="htc-job-file-transfer">📁 HTC Data guide</a> . 
```

Once your data, software, code, and HTCondor submit file are ready, we recommend you submit a few test jobs before running your full workflow.

Test jobs help check that your job is working correctly. They also help you understand how much CPU or GPU, memory, and disk space your job uses. This information can help you choose better resource requests later and troubleshoot problems before submitting many jobs.

Start with **one test job** first. After it finishes, check the files HTCondor creates to make sure the job ran correctly.

HTCondor creates several files that can help you review what happened when your test job ran:

<p style="text-align:center"><img src="/images/roadmap-output-files.png" alt="Icons and descriptions for .log, .out, and .err files" width=600px></p>

- The <code>.log</code> file shows information about how the job ran and resources used.
- The <code>.out</code> file shows the text your code printed while it was running.
- The <code>.err</code> file shows error messages from your software or code.

After your test jobs finish, we recommend you check for:

- Jobs that were placed on hold. You can view hold messages with <code>condor_q jobID -hold</code>.

> - Expected output files.

Reviewer: ```suggestion
- Expected output files - are they in the right location, and do they show the expected output?
```

> - The size and number of output files. This helps you make sure files are being saved in the correct location and that your quota is large enough for your output data when you submit more jobs.

Reviewer: ```suggestion
- The size and number of output files. This helps you make sure that your quota is large enough for your output data when you submit more jobs.
```

> If your first test job works as expected, run a small batch next, such as 5 to 10 jobs. This gives you a chance to test your workflow at a slightly larger scale before submitting everything.

Reviewer: ```suggestion
If your first test job works as expected, scale up your test by running a small batch next, such as 5 to 10 jobs. This gives you a chance to test your workflow at a slightly larger scale and ensure your pipeline works before submitting everything.
```

> When the small batch finishes, review the output files and check your available storage space, also called your **quota**. This will help you estimate whether you have enough space for the files your full workflow will create.

Reviewer: ```suggestion
When the small batch finishes, review the output files and [check your available storage space](check-quota), also called your **quota**. This will help you estimate whether you have enough space for the files your full workflow will create.
```

> If the test jobs finish successfully and your quota is large enough, you are ready to submit your full workflow!

Reviewer: ```suggestion
> If the test jobs finish successfully and your quota is large enough, you are ready to submit your full workflow!
```
Putting this in a block quote for emphasis.
